include/deal.II/matrix_free/fe_point_evaluation.h

```diff
@@ -953,20 +969,10 @@ FEPointEvaluation<n_components, dim, spacedim, Number>::evaluate(
   const ArrayView<const Number> &         solution_values,
   const EvaluationFlags::EvaluationFlags &evaluation_flag)
 {
-  const bool precomputed_mapping = mapping_info_on_the_fly.get() == nullptr;
-  if (precomputed_mapping)
-    {
-      unit_points =
-        mapping_info->get_unit_points(current_cell_index, current_face_number);
-
-      if (update_flags & update_values)
-        values.resize(unit_points.size(), numbers::signaling_nan<value_type>());
-      if (update_flags & update_gradients)
-        gradients.resize(unit_points.size(),
-                         numbers::signaling_nan<gradient_type>());
-    }
+  if (!is_reinitialized)
+    reinit(numbers::invalid_unsigned_int);
```

Comment:
@gassmoeller Could this be the problematic change which leads to #14984 ? Could you try to call `FEPointEvaluation::reinit(numbers::invalid_unsigned_int)` manually after `NonMatching::MappingInfo::reinit()`?